### PR TITLE
Change to new asdf resource entry point

### PR DESCRIPTION
**Pull request by WilliamJamieson**
This PR switches the datamodels to using the new `asdf.resources` entry-point for the datamodels schemas.  

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Comment by WilliamJamieson** [comment]
I'm not sure why there are test failures. I cannot reproduce these locally.

File: jwst/datamodels/integration.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import sys
```
Comment (stscieisenhamer) [inline jwst/datamodels/integration.py:1]: docstrings for module and function should be added.

Comment (WilliamJamieson) [reply jwst/datamodels/integration.py:5]: See 69ea324 for documentation.

Comment (stscieisenhamer) [reply jwst/datamodels/integration.py:5]: LGTM

**Review by stscieisenhamer — APPROVED** [review]

**Comment by hbushouse** [comment]
Is it important to try to include this PR in our next B9.0 release candidate, which is expected to be formed either today or next Monday?  What kind of risks does it entail, if we try to include it at the last minute for the B9.0 delivery?

**Comment by WilliamJamieson** [comment]
> Is it important to try to include this PR in our next B9.0 release candidate, which is expected to be formed either today or next Monday? What kind of risks does it entail, if we try to include it at the last minute for the B9.0 delivery?

I don't think this is critical. Though it would be nice, as it might help solve some of the weird install/upgrade issues that users are having (schemas going missing when they upgrade in their environment). It totally eliminates the old entry point for the datamodels so if jwst can still write files which use datamodels it should be fully functional.

**Comment by hbushouse** [comment]
Full regression test run is at https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/443/

**Pull request merged**